examples/convert_ieeg_to_bids.py

```diff
@@ -0,0 +1,213 @@
+
+3. Check the result and compare it with the standard
+
+4. Confirm that written iEEG coordinates are the same before
+:func:`write_raw_bids` was called
+
+The iEEG data will be written by :func:`write_raw_bids` with
+the addition of extra metadata elements in the following files:
+
+    * sidecar.json
+    * electrodes.tsv
+    * coord_system.json
+    * events.tsv
+    * channels.tsv
+
+Compared to EEG data, the main differences are within the
+coord_system and electrodes files.
+For more information on these files,
+refer to the iEEG-BIDS specification.
+"""
+
+# Authors: Adam Li <adam2392@gmail.com>
+# License: BSD (3-clause)
+
+import os
+import tempfile
+from pprint import pprint
+
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy.io import loadmat
+
+import mne
+from mne_bids import write_raw_bids, make_bids_basename, read_raw_bids
+from mne_bids.utils import print_dir_tree
+
+###############################################################################
+# Step 1: Download the data
+# -------------------------
+#
+# First, we need some data to work with. We will use the
+# data downloaded via MNE-Python's API.
+#
+# `<https://mne.tools/stable/generated/mne.datasets.misc.data_path>`_.
+#
+# Conveniently, there is already a data loading function available with
+# MNE-Python:
+
+# get MNE directory w/ example data
+mne_dir = mne.get_config('MNE_DATASETS_SAMPLE_PATH')
+
+# The electrode coords data are in the Matlab format: '.mat'.
+# This is easy to read in with :func:`scipy.io.loadmat` function.
+mat = loadmat(mne.datasets.misc.data_path() + '/ecog/sample_ecog.mat')
+ch_names = mat['ch_names'].tolist()
+ch_names = [x.strip() for x in ch_names]
+elec = mat['elec']  # electrode positions given in meters
+
+###############################################################################
+# Now we make a montage stating that the iEEG contacts are in MRI
+# coordinate system.
+montage = mne.channels.make_dig_montage(ch_pos=dict(zip(ch_names, elec)),
+                                        coord_frame='mri')
+print('Created %s channel positions' % len(ch_names))
+print(dict(zip(ch_names, elec)))
+
+###############################################################################
+# We also need to get some sample iEEG data,
+# so we will just generate random data from white noise.
+# Here is where you would use your own data if you had it.
+ieegdata = np.random.rand(len(ch_names), 1000)
+
+# We will create a :func:`mne.io.RawArray` object and
+# use the montage we created.
+info = mne.create_info(ch_names, 1000., 'ecog')
+raw = mne.io.RawArray(ieegdata, info)
+raw.set_montage(montage)
+
+###############################################################################
+# Let us confirm what our channel coordinates look like.
+
+# make a plot of the sensors in 2D plane, or on a 3D plot
+fig = plt.figure()
+ax2d = fig.add_subplot(111)
+raw.plot_sensors(ch_type='ecog', axes=ax2d)
```

Comment:
why do you need to create a figure explicitly? Doing

`raw.plot_sensors(ch_type='ecog')`

does not do the same?